java/statestore/src/main/java/sleeper/statestore/transactionlog/DynamoDBTransactionLogSnapshotStore.java

```diff
@@ -177,6 +181,43 @@ public LatestSnapshots getLatestSnapshots() {
         }
     }
 
+    public Stream<TransactionLogSnapshot> getOldestSnapshots() {
+        long expiryDate = timeSupplier.get().toEpochMilli() - expiryInDays.toMillis();
```

Comment:
It might be simpler to pass in the most recent time that we care about, rather than calculating it here?